### ISSUE TITLE
Add capability to train using PyTorch Lightning

### ISSUE THREAD

**Issue by mohaghighat** [nSightSurgical/yolov7#1]

**Comment by mohaghighat** [comment]
Would this be of any use https://github.com/Iywie/pl_YOLO?

**Comment by n-fallahinia** [comment]
@mohaghighat I don't think so. I looked into this a little while ago and realized that in this one, yolo models are implemented from scratch and not from the original yolo repo. So if we want get comparable results we must use the same model as we used before. Also, right now the training loop is working and the only issue is adding all the performance metrics to the logger. Pytorch Lightning requires metrics to be in the format of `torch.metrics` so I'm just inheriting from the `torch.metrics` and re-implementing those in new classes so that they can be logged by the Lightining.
